numpy/_core/_add_newdocs.py

```diff
@@ -2009,8 +2009,9 @@
     identifier '->' as well as the list of output subscript labels.
     This feature increases the flexibility of the function since
     summing can be disabled or forced when required. The call
-    ``np.einsum('i->', a)`` is like :py:func:`np.sum(a, axis=-1) <numpy.sum>`,
-    and ``np.einsum('ii->i', a)`` is like :py:func:`np.diag(a) <numpy.diag>`.
+    ``np.einsum('...i->...', a)`` is like
+    :py:func:`np.sum(a, axis=-1) <numpy.sum>`, and ``np.einsum('ii->i', a)``
+    is like :py:func:`np.diag(a) <numpy.diag>`.
```

Comment:
Good catch, but ellipses aren't discussed until the next paragraph. It would be confusing to use them just before they are explained. Also, it would be a bit inconsistent to fix the `sum` example but not the `diag` example. How about:
```suggestion
    ``np.einsum('i->', a)`` is like :py:func:`np.sum(a) <numpy.sum>`
    for 1-D arrays, and ``np.einsum('ii->i', a)``
    is like :py:func:`np.diag(a) <numpy.diag>` for 2-D arrays.
```

Reply:
About `np.diag`: good catch!

more accurately:
`np.einsum('ii->i', a)` is like `np.diag` for **SQUARE** matrices (not any 2-d arrays),
  because while `np.diag(a)` works for any `n x m` arrays, `np.einsum('ii->i', a)` complains:
> ValueError: dimensions in single operand for collapsing index 'i' don't match

---

About `np.einsum('...i->...', a)` vs  `np.einsum('i->', a)`
- I thought these ellipsis are inserted implicitly.. So I though explicit is better than implicit in the docs.. But maybe I'm wrong: maybe they are not used implicitly, and the notation 'i->' works for some other reason - I'm not very familiar with `np.einsum` tbh.. So I don't know which one is better: use ellipsis, or no and say only 1d array.